src/lrud-input/bubble-key-input.ts

```diff
@@ -58,10 +58,12 @@ export default function bubbleKey(focusTree: FocusStore, key: LRUDKey) {
       propagationStopped = true;
     }
 
-    const sourcingLeafId = focusH[0];
-    const targetNode: Node | undefined = state.nodes?.[sourcingLeafId];
+    const targetNodeId = reverseFocusHierarchy[0];
 
-    focusH.forEach((focusedNodeId) => {
+    // This is the equivalent of event.target within DOM events.
+    const targetNode: Node = state.nodes[targetNodeId] as Node;
```

Comment:
@LBRDan I noticed you had two defensive things going on here: checking that `state.nodes` is defined and that the returned node could be `undefined`. I think it's safe to assume that both exist, but I'm curious what you think.

Reply:
@jamesplease You’re right
As most of the current solutions for the spatial navigation problem, this one conceptually assumes that there is always a leaf node
The guard is added more for a runtime getState() call is executed before calling all the events on the node.

I've been too cautious about this maybe...